sdk/src/loom/Utils/Injector.ls

```diff
@@ -105,31 +139,69 @@ class Injector
     public function apply(target:Object, tagName:String="Inject"):void
     {
         var type = target.getType();
-        var targets:Vector.<InjectionTarget> = injectionTargetCache[type];
 
+        // Get all potential targets from the cache.
+        var targets:Vector.<InjectionTarget> = injectionTargetCache[type];
         if(!targets)
         {
+            // Cache miss, create entry.
             targets = new Vector.<InjectionTarget>();
             injectionTargetCache[type] = targets;
 
-            // otherwise we need to create injection targets
+            // We need to create injection targets...
+
+            // ...for fields...
             for(var i = 0; i<type.getFieldInfoCount(); i++)
             {
+                // Look up the field.
                 var fieldInfo = type.getFieldInfo(i);
+
+                // Get the metadata if present.
                 var inject = fieldInfo.getMetaInfo(tagName);
-                if(inject) {
-                    var t = new InjectionTarget();
-                    t.type = fieldInfo.getTypeInfo();
-                    t.property = fieldInfo.getName();
-                    t.fieldInfo = fieldInfo;
-                    t.id = inject.getAttribute("id");
+                if(!inject)
+                    continue;
```

Comment:
Should there be some sort of feedback if the injection fails for both fields and properties?